libraries/Microsoft.Bot.Builder/TurnContextServiceCollection.cs

```diff
@@ -46,95 +22,44 @@ public TurnContextServiceCollection()
-    /// </summary>
-    [Serializable]
-    public class ServiceKeyAlreadyRegisteredException : Exception
-    {
-        public ServiceKeyAlreadyRegisteredException(string key) : base($"A services is already registered with the specified key: {key}")
-        {
+            // note this can throw with a duplicate key
+            base.Add(key, service);
         }
 
-        protected ServiceKeyAlreadyRegisteredException(SerializationInfo info, StreamingContext context) : base(info, context)
+        public void Add<TService>(TService service) where TService : class
         {
+            Add(typeof(TService).FullName, service);
         }
-    }
 
-    /// <summary>
-    /// Provides a set of convienience methods that extend the behavior of any <see cref="ITurnContextServiceCollection"/>. 
-    /// </summary>
-    public static class TurnContextServiceCollectionExtensions
-    {
-        
-        /// <summary>
-        /// Add a service using its full type name as the key.
-        /// </summary>
-        /// <typeparam name="TService">The type of service to be added.</typeparam>
-        /// <param name="service">The service to add.</param>
-        public static void Add<TService>(this ITurnContextServiceCollection serviceCollection, TService service) where TService : class =>
-            serviceCollection.Add(typeof(TService).FullName, service);
-
-        /// <summary>
-        /// Get a service by type using its full type name as the key.
-        /// </summary>
-        /// <typeparam name="TService">The type of service to be retrieved.</typeparam>
-        /// <returns>The service stored under the specified key.</returns>
-        public static TService Get<TService>(this ITurnContextServiceCollection serviceCollection) where TService : class =>
-            serviceCollection.Get<TService>(typeof(TService).FullName);
-
-        /// <summary>
-        /// Returns all entries in the collection of a specified type.
-        /// </summary>
-        /// <typeparam name="TService">The type of service to be found.</typeparam>
-        /// <param name="serviceCollection">An <see cref="ITurnContextServiceCollection"/> to search for services in.</param>
-        /// <returns>All instances of the requested service currently stored in the collection.</returns>
-        public static IEnumerable<KeyValuePair<string, TService>> GetServices<TService>(this ITurnContextServiceCollection serviceCollection) where TService : class
+        public void Dispose()
         {
-            foreach (var entry in serviceCollection)
+            foreach(var entry in Values)
             {
-                if (entry.Value is TService service)
+                if(entry is IDisposable disposableService)
                 {
-                    yield return new KeyValuePair<string, TService>(entry.Key, service);
+                    disposableService.Dispose();
```

Comment:
If one dispose throws, the rest won't be cleaned up. 

Is this the right semantic? Or should there be try/catch here to do this regardless?